src/consensus/params.h

```diff
@@ -49,6 +49,13 @@ struct BIP9Deployment {
      *  process (which takes at least 3 BIP9 intervals). Only tests that specifically test the
      *  behaviour during activation cannot use this. */
     static constexpr int64_t ALWAYS_ACTIVE = -1;
+
+    /** this value is used for forks activated by master nodes.
+      * negative values means it is regular fork, no masternodes confirmation is needed.
+      * 0 means that there's no approval from masternodes yet.
+      * Otherwise it shows minimum height when miner's signals for this block can be assumed
+      */
+    mutable int64_t nMNActivationHeight{-1};
```

Comment:
Should move it closer to `nFalloffCoeff`. Also, why `mutable`?

Reply:
it's mutable because CChainParams is defined as a const:

```
/** 
 * Return the currently selected parameters. This won't change after app
 * startup, except for unit tests.
 */
const CChainParams &Params();
```

It doesn't mean to be changed except nMNActivationHeight that is actually changed when "EHF signal" is mined. Any ideas how to refactor/change it without creating an alternate for CChainParams that will work for EHF forks only?

Reply:
Why not have this information stored inside of CMNHFManager?

Reply:
Bcs it is difficult to get access to mnhfManager from GetStateFor() and related code without multiple circular dependencies and gloval variables. Let me think how to refactor it. Can I refactor it out of scope of this PR?